src/components/widgets/camera/services/WebrtcGortcCamera.vue

```diff
@@ -0,0 +1,98 @@
+<template>
+  <video
+    ref="streamingElement"
+    autoplay
+    playsinline
+    controls
+    muted
+    :style="cameraStyle"
+    :crossorigin="crossorigin"
+  />
+</template>
+
+<script lang="ts">
+import { Component, Ref, Mixins } from 'vue-property-decorator'
+import CameraMixin from '@/mixins/camera'
+
+@Component({})
+export default class WebrtcGortcCamera extends Mixins(CameraMixin) {
+  @Ref('streamingElement')
+  readonly cameraVideo!: HTMLVideoElement
+
+  pc: RTCPeerConnection | null = null
+  remoteId: string | null = null
+  ws: WebSocket | null = null
+
+  // webrtc player methods
+  // adapted from https://github.com/AlexxIT/go2rtc/blob/master/www/webrtc.html
+  // also adapted from https://github.com/mainsail-crew/mainsail/pull/1651
+
+  get url () {
+    const urlSearch = new URL(this.buildAbsoluteUrl(this.camera.urlStream || '')).search.toString()
+    const url = new URL('api/ws' + urlSearch, this.buildAbsoluteUrl(this.camera.urlStream || ''))
+    url.searchParams.set('media', 'video+audio')
+    // change protocol to ws
+    if (url.protocol === 'https') { url.protocol = 'wss' + ':' } else { url.protocol = 'ws' + ':' }
```

Comment:
```suggestion
    url.protocol = url.protocol === 'https'
      ? 'wss:'
      : 'ws:'
```

Reply:
When I use a 2 in the name. Then the component name mapping fails. It maps with spaces in there. That's why I had to lose the 2.